redirect/main.tf

```diff
@@ -0,0 +1,78 @@
+locals {
+  name_prefix = var.name_prefix != null ? var.name_prefix : "${var.project}-${var.environment}"
+
+  default_tags = {
+    Project     = var.project
+    Environment = var.environment
+  }
+
+  tags = merge(local.default_tags, var.tags)
+}
+
+resource "aws_s3_bucket" "redirect" {
+  count = var.create ? 1 : 0
+
+  bucket = var.bucket != null ? var.bucket : "${local.name_prefix}-redirect"
+  acl    = "public-read"
+  tags   = local.tags
+
+  website {
+    redirect_all_requests_to = var.target
+  }
+}
+
+locals {
+  website_endpoint = var.create ? aws_s3_bucket.redirect[0].website_endpoint : ""
+}
+
+resource "aws_cloudfront_distribution" "redirect" {
+  count = var.create ? 1 : 0
+
+  origin {
+    domain_name = local.website_endpoint
+    origin_id   = aws_s3_bucket.redirect[0].id
+
+    custom_origin_config {
+      http_port              = 80
+      https_port             = 443
+      origin_protocol_policy = "http-only"
+      origin_ssl_protocols   = ["TLSv1.2"]
+    }
+  }
+
+  aliases         = var.domains
+  enabled         = true
+  is_ipv6_enabled = true
+  price_class     = var.cloudfront_price_class
+
+  default_cache_behavior {
+    allowed_methods        = ["DELETE", "GET", "HEAD", "OPTIONS", "PATCH", "POST", "PUT"]
```

Comment:
I'm not sure if it makes sense to allow all these http methods since responding with a redirect to non-GET requests usually confuses clients, eg. often the client will follow the redirect by making a GET request to the specified location

Reply:
As discussed for methods other than handled by S3 bucket there will be 405 Method not allowed with empty response.